knowledge-base/getauthenticationstateasync-was-called-before-setauthenticationstate.md

```diff
@@ -0,0 +1,66 @@
+---
+title: GetAuthenticationStateAsync was called before SetAuthenticationState
+description: Such an error might occur when you try to authenticate the REST service 
+type: troubleshooting
+page_title: GetAuthenticationStateAsync was Called before SetAuthenticationState
+slug: getauthenticationstateasync-was-called-before-setauthenticationstate
+position: 
+tags: 
+ticketid: 1487514
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+Error 'GetAuthenticationStateAsync was called before SetAuthenticationState.' is thrown in a [Custom Report Source Resolver](../reporting/telerik-reporting-rest-custom-report-resolver) that uses **GetAuthenticationStateAsync**.
```

Comment:
Use SLUG 
{%slug telerikreporting/using-reports-in-applications/host-the-report-engine-remotely/telerik-reporting-rest-services/rest-service-report-source-resolver/how-to-implement-a-custom-report-source-resolver%}